frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -593,7 +579,24 @@ export function ConfidenceMetricsSection({
   filter,
 }: ConfidenceMetricsSectionProps) {
   const maxSelectedRocCurves: number = 10;
+  const [allLinkedArtifacts, setAllLinkedArtifacts] = useState<LinkedArtifact[]>(linkedArtifacts);
   const [linkedArtifactsPage, setLinkedArtifactsPage] = useState<LinkedArtifact[]>(linkedArtifacts);
+  const [filterString, setFilterString] = useState<string>('');
+
+  // Reload the page on linked artifacts refresh or re-selection.
+  useEffect(() => {
+    if (filter && !isEqual(linkedArtifacts, allLinkedArtifacts)) {
+      setLinkedArtifactsPage(linkedArtifacts);
+      setAllLinkedArtifacts(linkedArtifacts);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [linkedArtifacts]);
+
+  // Verify that the existing linked artifacts are correct; otherwise, wait for refresh.
+  if (filter && !isEqual(linkedArtifacts, allLinkedArtifacts)) {
+    return null;
+  }
```

Comment:
We add `eslint` warning disable as we only want to reload the page when the `linkedArtifacts` change from the original value (when runs are re-selected and the `linkedArtifacts` value changes.

The early `return null` prevents an error when this component tries to fetch run information which is no longer present on the `filter`, and the `useEffect` seems necessary to properly reload the component as a whole.

As shown in the screencast, there does not seem to be any visual delay for this re-render when runs are re-selected, so I believe this logic is sound, but please let me know if you have suggested changes!